components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java

```diff
@@ -112,9 +113,15 @@ public boolean handleRequest(MessageContext mc) {
             	tenantDomain = contextAndVersion[1];
             }
 
-            WebApp webApp = getWebApp(context, version);
-            String api = webApp.getId().getApiName();
-            String api_version = api + ":" + version;
+            WebApp webApp = null;
+            if (version != null) {
+                webApp = getWebApp(context, version);
+            } else {
+                webApp = getNonVersionedWebApp(context, tenantDomain);
+                version = webApp.getId().getVersion();
+            }
+            String appName = webApp.getId().getApiName();
+            String api_version = appName + ":" + version;
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/d4eba04341eb91c4b68b50cbb78987715b1a657d